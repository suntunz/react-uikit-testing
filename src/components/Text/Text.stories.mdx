import { Meta, Story } from '@storybook/addon-docs'
import { DesignTokenDocBlock } from 'storybook-design-token/dist/doc-blocks'

<Meta title='Design Tokens/Font Sizes' />

# Font Sizes

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Pariatur qui ut
beatae nam molestiae cum doloremque fugit nihil, vitae modi recusandae culpa
odit facere eum vel reiciendis eius repellat, totam aperiam. Molestiae error
voluptatibus et neque vel libero, harum nam quibusdam mollitia, sapiente rerum
iste modi sit deserunt? Officiis, quaerat!

<DesignTokenDocBlock categoryName='Font Sizes' />

## As cards

<DesignTokenDocBlock categoryName='Font Sizes' viewType='card' />
