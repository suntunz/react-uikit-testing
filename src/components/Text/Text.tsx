import React from 'react'

import switchCase from '../../utils/swtich-case'

export interface ITextProps {
    /**
     * text label
     */
    text: string

    /**
     * ####text size
     * - **tiny**: 10px;
     * - **xs**: 12px
     * - **sm**: 14px
     * - **base**: 16px
     * - **lg**: 18px
     * - **xl**: 20px
     * @default base
     */
    size?: 'tiny' | 'xs' | 'sm' | 'base' | 'lg' | 'xl'

    /**
     * text color
     * @default base
     */
    color?: 'base' | 'primary' | 'secondary' | 'red' | 'yellow'

    /**
     * font family
     * @default sans
     */
    fontFamily?: 'sans' | 'serif' | 'mono'

    /**
     * custom class name
     */
    className?: string
}

const Text: React.FC<ITextProps> = (props) => {
    const { text, size = 'base', color = 'base', fontFamily = '', className = null } = props

    const allowFontSize = ['tiny', 'xs', 'sm', 'base', 'lg', 'xl']
    const allowFontFamily = ['sans', 'serif', 'mono']
    const allowColor = ['base', 'primary', 'secondary', 'red', 'yellow']

    const fontSizeClass = allowFontSize.includes(size)
        ? switchCase(size, {
              tiny: 'text-tiny',
              xs: 'text-xs',
              sm: 'text-sm',
              base: 'text-base',
              lg: 'text-lg',
              xl: 'text-xl',
              default: 'text-base',
          })
        : ''

    const fontFamilyClass = allowFontFamily.includes(fontFamily)
        ? switchCase(fontFamily, {
              sans: 'font-sans',
              serif: 'font-serif',
              mono: 'font-mono',
              default: 'font-sans',
          })
        : ''

    const fontColorClass = allowColor.includes(color)
        ? switchCase(color, {
              base: 'text-color-base',
              primary: 'text-primary',
              secondary: 'text-secondary',
              red: 'text-red-400',
              yellow: 'text-amber-300',
              default: 'text-color-base',
          })
        : ''

    const classNames = [fontSizeClass, fontFamilyClass, fontColorClass, className].join(' ').trim()

    return <div className={classNames}>{text}</div>
}

export default Text
