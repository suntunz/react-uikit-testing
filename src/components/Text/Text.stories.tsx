import { Meta, Story } from '@storybook/react'

import Text, { ITextProps } from './Text'

export default {
    title: 'Typography/Text',
    component: Text,
    argTypes: {
        text: {
            control: false,
        },
        size: {
            control: false,
        },
        color: {
            control: false,
        },
        fontFamily: {
            control: false,
        },
        className: {
            control: false,
        },
    },
} as Meta

const Template: Story<ITextProps> = (args) => {
    return (
        <div>
            <Text {...args} size='tiny' text='Tiny size: 10px' />
            <Text {...args} size='xs' text='Extra small size: 12px' />
            <Text {...args} size='sm' text='Small size: 14px' />
            <Text {...args} size='base' text='Base size: 16px' />
            <Text {...args} size='lg' text='Large size: 18px' />
            <Text {...args} size='xl' text='Extra Size: 20px' />
            <br />
            <Text {...args} fontFamily='sans' size='xl' text='with font family font-sans' />
            <Text {...args} fontFamily='serif' size='xl' text='with font family font-serif' />
            <Text {...args} fontFamily='mono' size='xl' text='with font family font-mono' />
            <br />
            <Text {...args} color='base' text='With base color' />
            <Text {...args} color='red' size='xl' text='With red color' />
            <Text {...args} color='yellow' size='xl' text='With yellow color' />
            <Text {...args} color='primary' size='xl' text='With primary color' />
            <Text {...args} color='secondary' size='xl' text='With secondary color' />
            <br />
            <Text {...args} className='text-3xl' color='red' text='With custom classname' />
        </div>
    )
}

export const Default = Template.bind({})
Default.args = {}
