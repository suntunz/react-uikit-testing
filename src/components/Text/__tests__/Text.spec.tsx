import React from 'react'
import { create } from 'react-test-renderer'

import Text from '../Text'

describe('Text component', () => {
    it('should match previous snapshot', () => {
        const component = create(<Text text='Test' />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
