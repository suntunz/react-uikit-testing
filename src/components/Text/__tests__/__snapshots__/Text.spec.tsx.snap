// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should match previous snapshot 1`] = `
<div
  className="text-base"
>
  Test
</div>
`;
